### ISSUE TITLE
Versioning / Version check of adaptations

### ISSUE THREAD

**Issue by markusschloesser** [christofmuc/KnobKraft-orm#123]
When I installed and ran the new version from today, KK greeted me with "detected user adaptation, overriding inbuilt one" (or similar), which was the one for the sq80 which I downloaded from the discussion here on GH.
Now the inbuilt one would have been the newer one, that's why I deleted the user adaptation, so how about:
1. versioning for adaptations
2. notify user of there's a newer one inbuilt

**Comment by Mostelin** [comment]
Even though I knew what was happening, when I saw that message I thought that some users might interpret it as an error of some kind, so I agree that perhaps there is a better way to handle this.

Incidentally, the new version 0.9 merely translates names in a nicer way, and has some clean-up work in its comments.

**Comment by christofmuc** [comment]
@markusschloesser Agree, but this is a can of worms, really. I try to do the versioning for user-contributed adaptations, because you should really use git (hub) for this. And the versioning is not a straight line of versions, you might have started making changes on an earlier version, then a new version gets build in... this is essentially a fork requiring a 3-way merge instead of just being "newer". Scratching my head here.

**Comment by markusschloesser** [comment]
mmh, maybe keep it simple in the beginning and use the DATE as the indicator?

**Comment by christofmuc** [comment]
@markusschloesser So like "this version of KnobKraft has an adaptation that is newer than your local version" as a warning?

**Comment by markusschloesser** [comment]
Yes, exactly that.

Happy new year to you! 😊

**Comment by christofmuc** [comment]
@markusschloesser Happy new year :-)

**Comment by Andy2No** [comment]
@christofmuc @markusschloesser @Mostelin Happy new year! :)

**Comment by christofmuc** [comment]
@Andy2No Hihi, happy new year as well!
